### PR TITLE
bcc/perf: fix panic invalid memory address

### PR DESCRIPTION
**Pull request by schu**
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x556edd8ee96e]

```
goroutine 9 [running]:
panic(0x556edd94e940, 0xc42000c140)
        /usr/lib/go-1.7/src/runtime/panic.go:500 +0x1a1
github.com/iovisor/gobpf/bcc.callback_to_go(0xc42000c5c8, 0x7f45b4bee24c, 0xc400000114)
        /go/src/github.com/iovisor/gobpf/bcc/perf.go:92 +0x4e
github.com/iovisor/gobpf/bcc._cgoexpwrap_9ea30c17befe_callback_to_go(0xc42000c5c8, 0x7f45b4bee24c, 0xc400000114)
        ??:0 +0x3d
github.com/iovisor/gobpf/bcc._Cfunc_perf_reader_poll(0x556e00000004, 0xc420014220, 0xc4000001f4, 0x0)
        ??:0 +0x4d
github.com/iovisor/gobpf/bcc.(*PerfMap).poll(0xc420016540, 0x1f4)
        /go/src/github.com/iovisor/gobpf/bcc/perf.go:168 +0x13c
created by github.com/iovisor/gobpf/bcc.(*PerfMap).Start
        /go/src/github.com/iovisor/gobpf/bcc/perf.go:151 +0x48
```

Helped-by: Alban Crequy <alban@kinvolk.io>

### CODE REVIEW

**Comment by schu** [comment]
cc @alban @iaguis

File: bcc/perf.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ import (
 #include <bcc/libbpf.h>
 #include <bcc/perf_reader.h>
 
-extern void callback_to_go(uint64_t*, void*, int);
+extern void callback_to_go(void*, void*, int);
```
Comment (alban) [inline bcc/perf.go]: I would add a comment explaining where this prototype comes from:
```
src/cc/libbpf.h:45:typedef void (*perf_reader_raw_cb)(void *cb_cookie, void *raw, int raw_size);
```

File: bcc/perf.go [hunk 2]
```diff
@@ -87,8 +87,8 @@ func lookupCallback(i uint64) *callbackData {
 // function and give that to C-code since the cgo tool will generate a
 // stub in C that should be called."
 //export callback_to_go
-func callback_to_go(i *C.uint64_t, raw unsafe.Pointer, rawSize C.int) {
-	callbackData := lookupCallback(uint64(*i))
+func callback_to_go(i unsafe.Pointer, raw unsafe.Pointer, rawSize C.int) {
```
Comment (alban) [inline bcc/perf.go]: I'd change the variable name `i` to `cb_cookie` to match the prototype from bcc

Comment (alban) [inline bcc/perf.go]: or cbCookie in Go

File: bcc/perf.go (hunk 1 above)
Comment (schu) [inline bcc/perf.go]: Done.

File: bcc/perf.go (hunk 2 above)
Comment (schu) [inline bcc/perf.go]: Done.

**Comment by alban** [comment]
LGTM

**Pull request merged**